streams/jvm/src/main/scala/zio/stream/compression/package.scala

```diff
@@ -0,0 +1,89 @@
+package zio.stream
+
+import java.util.zip.{ DataFormatException, Inflater }
+import java.{ util => ju }
+
+import scala.annotation.tailrec
+
+import zio._
+
+package object compression {
+
+  /**
+   * Decompresses deflated stream. Compression method is described in https://tools.ietf.org/html/rfc1951.
+   *
+   * @param noWrap  wheater wrapped in ZLIB header and trailer, see https://tools.ietf.org/html/rfc1951.
+   *                For HTTP 'deflate' content-encoding should be false, see https://tools.ietf.org/html/rfc2616
+   * @param bufferSize size of buffer used internally
+   * HTTP 'deflate' content-encoding should use nowrap = false. See .
+   * */
+  def inflate(
+    bufferSize: Int = 64 * 1024,
+    noWrap: Boolean = false
+  ): ZTransducer[Any, CompressionException, Byte, Byte] = {
+    def makeInflater(
+      bufferSize: Int
+    ): ZManaged[Any, Nothing, Option[zio.Chunk[Byte]] => ZIO[Any, CompressionException, Chunk[Byte]]] =
+      ZManaged
+        .make(ZIO.effectTotal((new Array[Byte](bufferSize), new Inflater(noWrap)))) {
+          case (_, inflater) => ZIO.effectTotal(inflater.end())
+        }
+        .map {
+          case (buffer, inflater) => {
+            case None => ZIO.succeed(Chunk.empty)
```

Comment:
`pullOutput`, which I'll rename to `pullAllOutput` does it, when after reading some part of data it is `.finished()`. What is more if there are multiple delfated parts inside one stream, it would work. 
However it wouldn't work if not whole deflated part was transduced (`Inflater` wouldn't be `.finished()`). Thanks for catching this! I'll add test for this case and fix!

Reply:
@iravid actually I don't know if I can test it at ZStream level, should I write the test on level of calling `push`?